subsys/net/ip/net_if.c

```diff
@@ -4360,12 +4365,23 @@ static void init_igmp(struct net_if *iface)
 
 int net_if_up(struct net_if *iface)
 {
+	const struct device *dev;
 	int status = 0;
 
 	NET_DBG("iface %p", iface);
 
 	net_if_lock(iface);
 
+	/* If the device is not ready it is pointless trying to take it up. */
+	dev = net_if_get_device(iface);
+	NET_ASSERT(dev);
+
+	if (!device_is_ready(dev)) {
+		NET_DBG("Device %s (%p) is not ready", dev->name, dev);
+		status = -ENXIO;
+		goto out;
+	}
+
```

Comment:
After some more investigation, I am not sure if we should actually do this. If we return error at this point and not call device driver `start()` API function, then the device does not have a chance to try to take the device up.
Or we could move this check later in this function and only check the device readiness if the L2 does not implement the `enable()`, in which case the device does not have `start()` function what we could call.
WDYT?

Reply:
I think you're right, if the device is not ready, it should fail on `start()`